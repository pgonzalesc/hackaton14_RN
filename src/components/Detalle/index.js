import React, {useState} from 'react';
import {StyleSheet, View, Text, Image, Dimensions, TouchableOpacity} from 'react-native';
import Icon from 'react-native-vector-icons/Ionicons';
const {height, width} = Dimensions.get('window');
const Detalle = ( {navigation} )=>{
    const amount = 1;
    const unitPrice = 15;
    const [counter, setCounter] = useState(amount);
    const [price, setPrice] = useState(15);
    const [selectedSize, setSelectedSize] = useState({s:true, m:false, l:false});
    
    const handleCounter = (opc) => {
        if(opc === 'P') {//Plus
            setCounter( counter + 1 );
            setPrice( unitPrice*(counter+1) );
        }
        else if(opc === 'M') {//Minus

            if(counter - 1 >= 0) {
                setCounter( counter - 1 );
                setPrice( unitPrice*(counter-1) );
            }
            
        }
    }

    const handleSize = (size) => {
        if(size === 'S') {
            setCounter(amount);
            setPrice(unitPrice);
            setSelectedSize({s:true, m:false, l:false});
        }
        else if(size === 'M') {
            setCounter(amount);
            setPrice(unitPrice+3);
            setSelectedSize({s:false, m:true, l:false});
        }
        else if(size === 'L') {
            setCounter(amount);
            setPrice(unitPrice+5);
            setSelectedSize({s:false, m:false, l:true});
        }
            
    }

    return (
        <View style={styles.container}>
            <View style={{marginHorizontal: 20, marginVertical: 15}}>
                <TouchableOpacity onPress={()=>{navigation.navigate('Home');}}>
                    <Icon name={'md-arrow-back-outline'} size={30} color={'#e1e1e1'}/>
                </TouchableOpacity>
            </View>
            <View style={styles.containerTitle}>
                <Text style={styles.textTitle}>Americano Coffee</Text>
            </View>
            <View style={styles.containerPrice}>
                <Text style={styles.textPrice}>${price}</Text>
            </View>
            <View style={{flexDirection: 'row', alignItems: 'center', justifyContent: 'center'}}>
                <View style={{marginHorizontal: 20}}>
                    {selectedSize.s 
                     ?  <TouchableOpacity style={styles.buttonSelected} onPress={()=>handleSize('S')}> 
                            <Text style={styles.textButtonSelected}>S</Text>
                        </TouchableOpacity>
                     :  <TouchableOpacity style={styles.button} onPress={()=>handleSize('S')}>
                            <Text style={styles.textButton}>S</Text>
                        </TouchableOpacity>
                    }
                    {selectedSize.m
                     ?  <TouchableOpacity style={styles.buttonSelected} onPress={()=>handleSize('M')}> 
                            <Text style={styles.textButtonSelected}>M</Text>
                        </TouchableOpacity>
                     :  <TouchableOpacity style={styles.button} onPress={()=>handleSize('M')}>
                            <Text style={styles.textButton}>M</Text>
                        </TouchableOpacity>
                    }
                    {selectedSize.l
                     ?  <TouchableOpacity style={styles.buttonSelected} onPress={()=>handleSize('L')}> 
                            <Text style={styles.textButtonSelected}>L</Text>
                        </TouchableOpacity>
                     :  <TouchableOpacity style={styles.button} onPress={()=>handleSize('L')}>
                            <Text style={styles.textButton}>L</Text>
                        </TouchableOpacity>
                    }
                </View>
                <Image 
                 style={styles.image}
                 source={{uri:'https://www.starbucks.com.pe/media/cappucino-espresso_tcm92-18666_w1024_n.png'}}
                />
            </View>
            <View style={{flexDirection: 'row', alignItems: 'center', justifyContent: 'center'}}>
                <TouchableOpacity style={styles.buttonRounded} onPress={()=>handleCounter('P')}>
                    <Text style={styles.textSign}>+</Text>
                </TouchableOpacity>
                <View style={styles.containerAmount}>
                    <Text style={styles.textAmount}>{counter}</Text>
                </View>
                <TouchableOpacity style={styles.buttonRounded} onPress={()=>handleCounter('M')}>
                    <Text style={styles.textSign}>-</Text>
                </TouchableOpacity>
            </View>
            <View style={{alignItems: 'center'}}>
                <TouchableOpacity>
                    <View style={{backgroundColor:'#e1e1e1', paddingHorizontal: 50, paddingVertical: 10, borderRadius: 20}}>
                        <Text style={{color: '#376363',fontWeight: 'bold'}}>Add to cart</Text>
                    </View>
                </TouchableOpacity>
            </View>
        </View>
    )
}

const styles = StyleSheet.create({
    container: {
      flex: 1,
      backgroundColor: '#3E706F',
    },
    image: {
        width: width/2,
        height: height / 3,
    },
    containerTitle: {
        marginTop: 50,
    },
    textTitle: {
        color: '#e1e1e1',
        textAlign: 'center',
        fontSize: 20,
        fontWeight: 'bold',
    },
    containerPrice: {
        marginTop: 10,
        marginBottom: 20,
    },
    textPrice: {
        color: '#D5E46D',
        textAlign: 'center',
        fontSize: 22,
        fontWeight: 'bold',
    },
    button: {
        paddingVertical:10, 
        paddingHorizontal:14, 
        borderWidth:0.3, 
        borderColor: '#e1e1e1',
        borderRadius: 8,
        marginVertical: 10,
        backgroundColor: '#376363',
    },
    buttonSelected: {
        paddingVertical:10, 
        paddingHorizontal:14, 
        borderWidth:0.3, 
        borderColor: '#e1e1e1',
        borderRadius: 8,
        marginVertical: 10,
        backgroundColor: '#B6BF62',
    },
    textButton: {
        color: '#e1e1e1',
        textAlign: 'center',
        fontSize: 15,
        fontWeight: 'bold',
    },
    textButtonSelected:{
        color: '#376363',
        textAlign: 'center',
        fontSize: 15,
        fontWeight: 'bold',
    },
    buttonRounded:{
        borderWidth:1, 
        borderColor: '#e1e1e1', 
        paddingHorizontal: 15, 
        paddingVertical:8,
        borderRadius:20
    },
    containerAmount: {
        marginVertical: 40,
        paddingHorizontal: 30,
    },
    textAmount: {
        color: '#e1e1e1',
        fontSize: 25,
        fontWeight: 'bold',
    },
    textSign: {
        color: '#e1e1e1',
        fontSize: 20,
        fontWeight: 'bold',
    }
});

export default Detalle;