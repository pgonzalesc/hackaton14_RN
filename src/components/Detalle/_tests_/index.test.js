import React from 'react';
import { TouchableOpacity, View } from 'react-native';
import {shallow} from 'enzyme';
import Detalle from '../index';

const onPress = jest.fn();

describe('Pruebas en <Detalle/>', () => {
    let wrapper = shallow(<Detalle/>);

    beforeEach(()=>{
        wrapper = shallow(<Detalle/>);
    });

    test('Debe de mostrar <Detalle/> correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe seleccionar Tamaño S', () => {
        expect(onPress).not.toHaveBeenCalled();
        wrapper.find(TouchableOpacity).at(1).simulate('press');
        const value = wrapper.find(TouchableOpacity).at(1).children().children().text();
        expect(value).toBe('S');
    });

    test('Debe seleccionar Tamaño M', () => {
        expect(onPress).not.toHaveBeenCalled();
        wrapper.find(TouchableOpacity).at(2).simulate('press');
        const value = wrapper.find(TouchableOpacity).at(2).children().children().text();
        expect(value).toBe('M');
    });

    test('Debe seleccionar Tamaño L', () => {
        expect(onPress).not.toHaveBeenCalled();
        wrapper.find(TouchableOpacity).at(3).simulate('press');
        const value = wrapper.find(TouchableOpacity).at(3).children().children().text();
        expect(value).toBe('L');
    });

    test('Debe incrementar la cantidad en uno', () => {
        expect(onPress).not.toHaveBeenCalled();
        wrapper.find(TouchableOpacity).at(4).simulate('press');
        const value = wrapper.find(View).children().at(4).children().at(1).children().children().text();
        expect(value).toBe("2");
    });
    
    test('Debe disminuir la cantidad en uno', () => {
        expect(onPress).not.toHaveBeenCalled();
        wrapper.find(TouchableOpacity).at(4).simulate('press');
        wrapper.find(TouchableOpacity).at(5).simulate('press');
        const value = wrapper.find(View).children().at(4).children().at(1).children().children().text();
        expect(value).toBe("1");
    });
});
