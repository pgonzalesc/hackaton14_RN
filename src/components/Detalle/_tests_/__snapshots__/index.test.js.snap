// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <Detalle/> Debe de mostrar <Detalle/> correctamente 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#3E706F",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "marginHorizontal": 20,
        "marginVertical": 15,
      }
    }
  >
    <ForwardRef
      onPress={[Function]}
    >
      <Icon
        allowFontScaling={false}
        color="#e1e1e1"
        name="md-arrow-back-outline"
        size={30}
      />
    </ForwardRef>
  </View>
  <View
    style={
      Object {
        "marginTop": 50,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#e1e1e1",
          "fontSize": 20,
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
    >
      Americano Coffee
    </Text>
  </View>
  <View
    style={
      Object {
        "marginBottom": 20,
        "marginTop": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#D5E46D",
          "fontSize": 22,
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
    >
      $
      15
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "marginHorizontal": 20,
        }
      }
    >
      <ForwardRef
        onPress={[Function]}
        style={
          Object {
            "backgroundColor": "#B6BF62",
            "borderColor": "#e1e1e1",
            "borderRadius": 8,
            "borderWidth": 0.3,
            "marginVertical": 10,
            "paddingHorizontal": 14,
            "paddingVertical": 10,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#376363",
              "fontSize": 15,
              "fontWeight": "bold",
              "textAlign": "center",
            }
          }
        >
          S
        </Text>
      </ForwardRef>
      <ForwardRef
        onPress={[Function]}
        style={
          Object {
            "backgroundColor": "#376363",
            "borderColor": "#e1e1e1",
            "borderRadius": 8,
            "borderWidth": 0.3,
            "marginVertical": 10,
            "paddingHorizontal": 14,
            "paddingVertical": 10,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#e1e1e1",
              "fontSize": 15,
              "fontWeight": "bold",
              "textAlign": "center",
            }
          }
        >
          M
        </Text>
      </ForwardRef>
      <ForwardRef
        onPress={[Function]}
        style={
          Object {
            "backgroundColor": "#376363",
            "borderColor": "#e1e1e1",
            "borderRadius": 8,
            "borderWidth": 0.3,
            "marginVertical": 10,
            "paddingHorizontal": 14,
            "paddingVertical": 10,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#e1e1e1",
              "fontSize": 15,
              "fontWeight": "bold",
              "textAlign": "center",
            }
          }
        >
          L
        </Text>
      </ForwardRef>
    </View>
    <Image
      source={
        Object {
          "uri": "https://www.starbucks.com.pe/media/cappucino-espresso_tcm92-18666_w1024_n.png",
        }
      }
      style={
        Object {
          "height": 444.6666666666667,
          "width": 375,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <ForwardRef
      onPress={[Function]}
      style={
        Object {
          "borderColor": "#e1e1e1",
          "borderRadius": 20,
          "borderWidth": 1,
          "paddingHorizontal": 15,
          "paddingVertical": 8,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#e1e1e1",
            "fontSize": 20,
            "fontWeight": "bold",
          }
        }
      >
        +
      </Text>
    </ForwardRef>
    <View
      style={
        Object {
          "marginVertical": 40,
          "paddingHorizontal": 30,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#e1e1e1",
            "fontSize": 25,
            "fontWeight": "bold",
          }
        }
      >
        1
      </Text>
    </View>
    <ForwardRef
      onPress={[Function]}
      style={
        Object {
          "borderColor": "#e1e1e1",
          "borderRadius": 20,
          "borderWidth": 1,
          "paddingHorizontal": 15,
          "paddingVertical": 8,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#e1e1e1",
            "fontSize": 20,
            "fontWeight": "bold",
          }
        }
      >
        -
      </Text>
    </ForwardRef>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
      }
    }
  >
    <ForwardRef>
      <View
        style={
          Object {
            "backgroundColor": "#e1e1e1",
            "borderRadius": 20,
            "paddingHorizontal": 50,
            "paddingVertical": 10,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#376363",
              "fontWeight": "bold",
            }
          }
        >
          Add to cart
        </Text>
      </View>
    </ForwardRef>
  </View>
</View>
`;
